app/models/base_rule.rb

```diff
@@ -14,6 +14,18 @@ class BaseRule < ApplicationRecord
   has_many :checks, dependent: :destroy
   has_many :references, dependent: :destroy
 
+  has_and_belongs_to_many :satisfied_by,
+                          class_name: "BaseRule",
+                          join_table: :rule_satisfactions,
+                          foreign_key: :base_rule_id,
+                          association_foreign_key: :satisfied_by_base_rule_id
+
+  has_and_belongs_to_many :satisfies,
+                          class_name: "BaseRule",
+                          join_table: :rule_satisfactions,
+                          foreign_key: :satisfied_by_base_rule_id,
+                          association_foreign_key: :base_rule_id
+
```

Comment:
These should probably only be defined for `Rule`s, not `BaseRule`s since BaseRules will never have these relationships.

Reply:
https://github.com/mitre/vulcan/pull/290/commits/bd4b9de79290e46d47d7738ffa5d272d7c0d9e92